### PR TITLE
fix zui blocking untill services are up

### PR DESCRIPTION
**Pull request by Eslam-Nawara**
### Description

Fix `zui` to show the state of services while starting them instead of black screen.

### Related Issues
- #2263 

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: cmds/modules/zui/service.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	if _, err := register.NodeID(ctx); err != nil {
+		if isInProgressError(err) {
+			return InProgressStatus
+		} else {
+			return red(FailedStatus)
+		}
+	} else {
+		return green(activeStatus)
+	}
+}
```
Comment (ashraffouda) [inline cmds/modules/zui/service.go:132]: wouldn't be a bit clean if we removed those else statements? to be like this 
```
func getRegistrarStatus(ctx context.Context, client zbus.Client) string {
	register := stubs.NewRegistrarStub(client)
	if _, err := register.NodeID(ctx); err != nil {
		if isInProgressError(err) {
			return InProgressStatus
		}
		return red(FailedStatus)
	}
	return green(activeStatus)

}
```

**Review by ashraffouda — CHANGES_REQUESTED** [review]

File: cmds/modules/zui/service.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		{"Noded", InactiveStatus},
+		{"Powerd", InactiveStatus},
+	}
+
+	render.Signal()
+
+	done := make(chan bool)
+
+	go func() {
+		networkStatus := make(chan string)
```
Comment (ashraffouda) [inline cmds/modules/zui/service.go]: isn't it better if we have only one chan that all services write to it and we check the returned string against which service is up or so?

**Review by ashraffouda — CHANGES_REQUESTED** [review]

File: cmds/modules/zui/service.go (hunk 2 above)
Comment (muhamadazmy) [reply cmds/modules/zui/service.go]: I agree

**Review by muhamadazmy — COMMENTED** [review]
Really impressive work @Eslam-Nawara The behaviour of ZUI is much better now and i like how it looks as well.

I agree with Ashraf on his comment, please check it out so we can merge. What he basically suggest is we use a single channel that has status and service name not just a bool.

File: cmds/modules/zui/prov.go [hunk 3]
```diff
@@ -45,11 +46,11 @@ func provRender(client zbus.Client, render *signalFlag, prov *widgets.Table) err
 
 	prov.Rows = [][]string{
 		{"", "Total", "Reserved"},
-		{"CRU", "", ""},
-		{"Memory", "", ""},
-		{"SSD", "", ""},
-		{"HDD", "", ""},
-		{"IPv4", "", ""},
+		{"CRU", "Loading...", "Loading..."},
```
Comment (xmonader) [inline cmds/modules/zui/prov.go]: extract the `Loading ...` literal into a string and reuse it for consistency

**Review by xmonader — APPROVED** [review]

**Review by ashraffouda — APPROVED** [review]

**Pull request merged**